### ISSUE TITLE
remote_file should allow idempotence

### ISSUE THREAD

**Issue by paustin01** [widgit-cookbooks/nxlog#10]
Thanks for your great work on this cookbook.  Had run chef on a couple Windows nodes today and must have hit a slow mirror because it added several minutes to the run while trying to download the msi on servers that already had it installed and in the cache at the same version.  Seems as though nxlog isn't supporting etags and the default use_conditional_get isn't providing any idempotence.  Bummer to have to download on every run.  Seems like maybe it would be good to do a conditional checksum option.  E.g. 

```
if node['nxlog']['installer_checksum']
  remote_file 'nxlog' do
    path "#{Chef::Config[:file_cache_path]}/#{package_name}"
    source "http://nxlog.org/system/files/products/files/1/#{package_name}"
    checksum node['nxlog']['installer_checksum']
    mode 0644
  end
else
  remote_file 'nxlog' do
    path "#{Chef::Config[:file_cache_path]}/#{package_name}"
    source "http://nxlog.org/system/files/products/files/1/#{package_name}"
    mode 0644
  end
end
```

Where node['nxlog']['installer_checksum'] would be defined in the various platform recipes with a default value of nil.  Not the most elegant solution so open to feedback but I can do a PR if sounds ok.

**Comment by gh2k** [comment]
It's supposed to use etags or if-modified-since, at least on Chef 12...

This change is a good idea though. I don't see any reason why we couldn't also provide checksums for the default version specified in the attributes. Users would only have to change the checksum if they also change the version number.

**Comment by gh2k** [comment]
Released as 0.8.2
